nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/GetSNMP.java

```diff
@@ -87,113 +91,110 @@
             .description("All FlowFiles that cannot received from the SNMP agent are routed to this relationship")
             .build();
 
-    /** list of property descriptors */
-    private final static List<PropertyDescriptor> propertyDescriptors;
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = createPropertyList();
 
-    /** list of relationships */
-    private final static Set<Relationship> relationships;
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
 
-    /*
-     * Will ensure that the list of property descriptors is build only once.
-     * Will also create a Set of relationships
-     */
-    static {
-        List<PropertyDescriptor> _propertyDescriptors = new ArrayList<>();
-        _propertyDescriptors.add(OID);
-        _propertyDescriptors.add(TEXTUAL_OID);
-        _propertyDescriptors.add(SNMP_STRATEGY);
-        _propertyDescriptors.addAll(descriptors);
-        propertyDescriptors = Collections.unmodifiableList(_propertyDescriptors);
-
-        Set<Relationship> _relationships = new HashSet<>();
-        _relationships.add(REL_SUCCESS);
-        _relationships.add(REL_FAILURE);
-        relationships = Collections.unmodifiableSet(_relationships);
+    private SNMPGetter snmpGetter;
+
+    @OnScheduled
+    @Override
+    public void initSnmpClient(ProcessContext context) {
+        super.initSnmpClient(context);
+        String oid = context.getProperty(OID).getValue();
+        snmpGetter = new SNMPGetter(snmpContext.getSnmp(), snmpContext.getTarget(), new OID(oid));
     }
 
     /**
      * Delegate method to supplement
      * {@link #onTrigger(ProcessContext, ProcessSession)}. It is implemented by
      * sub-classes to perform {@link Processor} specific functionality.
      *
-     * @param context
-     *            instance of {@link ProcessContext}
-     * @param processSession
-     *            instance of {@link ProcessSession}
+     * @param context        instance of {@link ProcessContext}
+     * @param processSession instance of {@link ProcessSession}
      * @throws ProcessException Process exception
      */
     @Override
-    protected void onTriggerSnmp(ProcessContext context, ProcessSession processSession) throws ProcessException {
-        if("GET".equals(context.getProperty(SNMP_STRATEGY).getValue())) {
-            final ResponseEvent response = this.targetResource.get();
-            if (response.getResponse() != null){
-                FlowFile flowFile = processSession.create();
+    public void onTrigger(ProcessContext context, ProcessSession processSession) {
+        final String targetUri = snmpContext.getTarget().getAddress().toString();
+        final String snmpStrategy = context.getProperty(SNMP_STRATEGY).getValue();
+        final String oid = context.getProperty(OID).getValue();
+
+        if ("GET".equals(snmpStrategy)) {
```

Comment:
With the `SNMP_STRATEGY` taking a list of allowable values, it would be helpful to define `GET` and `WALK` as either static variables or perhaps create an enum for reuse in both the property definition and in this method.